### PR TITLE
[mvp-002]: add auth

### PR DESCRIPTION
**Pull request by coriandar**

### CODE REVIEW

File: frontend/src/contexts/AuthContext.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+export const AuthProvider: React.FC<{ children: React.ReactNode }> = ({
+    children,
+}) => {
+    const [isAuthenticated, setIsAuthenticated] = React.useState(false);
+    const [isLoading, setIsLoading] = React.useState(true);
+
+    React.useEffect(() => {
+        async function fetchAuthStatus() {
+            const res = await apiValidateToken("");
```
Comment (pendevx) [inline frontend/src/contexts/AuthContext.tsx:26]: Was this intentional, to send an empty string as the token?

Comment (coriandar) [reply frontend/src/contexts/AuthContext.tsx:26]: Yep, it's for the middleware.ts to also call the validate token. NextRequest, gets cookie and parses the idToken from it.

Comment (pendevx) [reply frontend/src/contexts/AuthContext.tsx:26]: Gotcha

**Comment by pendevx** [comment]
lgtm

**Review by pendevx — APPROVED** [review]

**Pull request merged**